lib/Parser/Parser.cpp

```diff
@@ -203,6 +205,25 @@ Token Lexer::getNextToken(std::string &ErrStr) {
                         StringRef(NumBegin, NumEnd - NumBegin), 10)};
   }
 
+  if (*Begin == '(') {
+    ++Begin;
+    const char *NumBegin = Begin;
+    while (*Begin == '0' || *Begin == '1' || *Begin == 'x')
+      ++Begin;
+    const char *NumEnd = Begin - 1;
```

Comment:
you only use this variable once-- rather just eliminate it and refer to Begin - 1 later on